src/status_im/protocol/web3/delivery.cljs

```diff
@@ -85,23 +87,33 @@
 (s/def :delivery/pending-message
   (s/keys :req-un [:message/sig :message/to :shh/payload :payload/ack? ::id
                    :message/requires-ack? :message/topic ::attempts ::was-sent?]
-          :opt-un [:message/pubKey :message/symKeyID]))
+          :opt-un [:message/pub-key :message/sym-key-password]))
 
-(defn add-prepared-pending-message!
-  [web3 {:keys [message-id to sym-key-id pub-key] :as pending-message}]
-  {:pre [(valid? :delivery/pending-message pending-message)]}
-  (debug :add-prepared-pending-message!)
-  (let [message          (assoc
-                           (select-keys pending-message [:sig :topic :payload])
-                           :symKeyID sym-key-id
-                           :pubKey pub-key)
-        pending-message' (assoc pending-message :message message
+(defn- do-add-pending-message!
+  [web3 {:keys [message-id to pub-key sym-key-id] :as pending-message}]
+  (let [message          (select-keys pending-message [:sig :topic :payload])
+        message'         (if sym-key-id
+                           (assoc message :symKeyId sym-key-id)
+                           (assoc message :pubKey pub-key))
+        pending-message' (assoc pending-message :message message'
                                                 :id message-id)]
     (swap! messages assoc-in [web3 message-id to] pending-message')
     (when to
       (swap! recipient->pending-message
              update to set/union #{[web3 message-id to]}))))
 
+(defn add-prepared-pending-message!
+  [web3 {:keys [sym-key-password] :as pending-message}]
+  {:pre [(valid? :delivery/pending-message pending-message)]}
+  (debug :add-prepared-pending-message!)
+  (if sym-key-password
```

Comment:
Could you add a comment detailing this conditional and potential impacts for caller?

Reply:
What comment? Like "If we have password for symmetric key let's generate it"? Not sure what exactly you want to see here.

Reply:
Is the call to `do-add-pending-message!` sync in both cases? How slow is that?
Not sure it matters.

Reply:
`do-add-pending-message!` this call is sync always, `shh-keys/get-sym-key` makes async call to RPC, but this call is fast, it doesn't depend on blockchain state or anything related, so it will not be blocked because of some magic reason.